docs/source/error_code_list.rst

```diff
@@ -0,0 +1,502 @@
+         print('hello', name)
+
+    greet('jack')  # OK
+    greet('jill', 'jack')  # Error: Too many arguments for "greet"  [call-arg]
+
+Check argument types [arg-type]
+-------------------------------
+
+Mypy checks that argument types in a call match the declared argument
+types in the signature of the called function (if one exists).
+
+Example:
+
+.. code-block:: python
+
+   from typing import List, Optional
+
+   def first(x: List[int]) -> Optional[int]:
+        return x[0] if x else 0
+
+   t = (5, 4)
+   # Error: Argument 1 to "first" has incompatible type "Tuple[int, int]";
+   #        expected "List[int]"  [arg-type]
+   print(first(t))
+
+Check calls to overloaded functions [call-overload]
+---------------------------------------------------
+
+When you call an overloaded function, mypy checks that at least one of
+the signatures of the overload items match the argument types in the
+call.
+
+Check validity of types [valid-type]
+------------------------------------
+
+Mypy checks that each type annotation and any expression that
+represents a type is a valid type. Examples of valid types include
+classes, union types, callable types, type aliases, and literal types.
+Examples of invalid types include bare integer literals, functions,
+variables, and modules.
+
+Require annotation if variable type is unclear [var-annotated]
+--------------------------------------------------------------
+
+In some cases mypy can't infer the type of a variable without an
+explicit annotation. Mypy treats this as an error. This often happens
+when you initialize a variable with an empty collection, and mypy
+can't infer the collection item type. Mypy replaces any parts of the
+type it couldn't infer with ``Any``.
+
+Example with an error:
+
+.. code-block:: python
+
+   class Bundle:
+       def __init__(self) -> None:
+           # Error: Need type annotation for 'items'
+           #        (hint: "items: List[<type>] = ...")  [var-annotated]
+           self.items = []
+
+   reveal_type(Bundle().items)  # list[Any]
+
+In this example we have an explicit annotation to silence the error:
+
+.. code-block:: python
+
+   from typing import List
+
+   class Bundle:
+       def __init__(self) -> None:
+           self.items: List[str] = []  # OK
+
+   reveal_type(Bundle().items)  # list[str]
+
+Check validity of overrides [override]
+--------------------------------------
+
+Mypy checks that an overridden method or attribute is compatible with
+the base class.  A method in a subclass must accept all arguments
+that the base class method accepts, and the return type must conform
+to the return type in the base class (Liskov substitution principle).
+
+Argument typess can be more general is a subclass (i.e., they can vary
+contravariantly).  The return type can be narrowed in a subclass
+(i.e., it can vary covariantly).  It's okay to define additional
+arguments in a subclass method, as long all extra arguments have default
+values or can be left out.
```

Comment:
What other mechanism is there to specify that an argument can be left out? Is this a reference to overloading?